cmake/app/boilerplate.cmake

```diff
@@ -27,13 +27,11 @@ if(uname_output MATCHES "MSYS")
   set(MSYS 1)
 endif()
 
-# CMake version 3.8.2 is the minimum supported version. Version 3.9 is
-# supported but it introduced a warning that we do not wish to
-# show to users. Specifically, it displays a warning when an OLD
-# policy is used, but we need policy CMP0000 set to OLD to avoid
-# copy-pasting cmake_minimum_required across application
+# CMake version 3.11 is the minimum supported version. As of Version 3.9
+# a warning is shown if an OLD  policy is used, but we need policy CMP0000
+# set to OLD to avoid copy-pasting cmake_minimum_required across application
 # CMakeLists.txt files.
-cmake_minimum_required(VERSION 3.8.2)
+cmake_minimum_required(VERSION 3.11)
```

Comment:
3.11 is too recent unfortunately.

The most recent Ubuntu LTS, 18.04, has 3.10.2 in it's package repository.

Using a CMake version that is not easily available makes it more difficult to get started with Zephyr.

FetchModule can be emulated through recursively invoking CMake, not pretty, but it
is the better solution for our users.

Reply:
Actually this is a part I always have been surprised about. The zephyr documentation mentions the use of pip and on the other hand manual installation of cmake while the latest cmake version is available through pip (https://pypi.org/project/cmake/). 
E.g. the only thing I had to do was "pip install --upgrade cmake" and I had 3.11 available on Ubuntu LTS 16.04.

Reply:
I was not aware that one could install non-python packages through pip.

It's not obvious to me that we would want to encourage this in general.
Generally speaking users (rightly IMHO) prefer the distribution
package managers over pip.

But perhaps for CMake this makes sense ...

In any case using pip would require a separate PR.

Reply:
I have no problem with adding cmake to our pip requirements.txt, we already require a ton of stuff in it.

Reply:
We need to check whether this would work on Windows, and the documentation should be updated. I am somewhat skeptical personally, especially given that, if users already have a copy of CMake on their system, this might install a second one.

Reply:
That said I am open to this if it really means an improvement for our users and works reliably on Linux, macOS and Windows. It's an interesting discovery actually @vanwinkeljan, I had no idea one could do this and it might end up being a good thing indeed.

Reply:
A general remark here is that for python it is best practice to install project dependencies in a dedicated virtual environment and not system wide (either through pip or a packet manager), in this way the system does not get polluted with unnecessary installs. Now the nice thing is if you install cmake through pip into a dedicated virtual environment it can coexist with a already system wide installed version.